### PR TITLE
[MCO] Bump stolostron builder to Go 1.21

### PR DESCRIPTION
**Pull request by douglascamata**

### CODE REVIEW

File: ci-operator/config/stolostron/multicluster-observability-operator/stolostron-multicluster-observability-operator-main.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ base_images:
     name: ubi-minimal
     namespace: ocp
     tag: "8"
-  stolostron_builder_go1.20-linux:
+  stolostron_builder_go1.21-linux:
     name: builder
     namespace: stolostron
     tag: go1.20-linux
```
Comment (thibaultmg) [inline ci-operator/config/stolostron/multicluster-observability-operator/stolostron-multicluster-observability-operator-main.yaml:9]: Shouldn't we update this tag?

Comment (douglascamata) [reply ci-operator/config/stolostron/multicluster-observability-operator/stolostron-multicluster-observability-operator-main.yaml]: Good catch. Just pushed a fix.

**Review by thibaultmg — APPROVED** [review]
/lgtm

**Pull request merged**